tests/foreman/installer/test_installer.py

```diff
@@ -1390,6 +1393,45 @@ def test_installer_options_and_sections(filter):
     assert previous == current, msg
 
 
+@pytest.mark.e2e
+@pytest.mark.tier1
+@pytest.mark.parametrize("sat_ready_rhel", [settings.server.version.rhel_version], indirect=True)
+def test_satellite_installation(sat_ready_rhel):
+    """Run a basic Satellite installation
+
+    :id: bbab30a6-6861-494f-96dd-23b883c2c906
+
+    :steps:
+        1. Get a RHEL host
+        2. Configure repos
+        3. Enable satellite module
+        4. Install satellite
+        5. Run satellite-installer
+
+    :expectedresults:
+        1. Correct satellite packaged is installed
+        2. satellite-installer runs successfully
+        3. Health check runs successfully
+
+    :CaseImportance: High
+    """
+    sat_version = settings.server.version.release
+    # Register for RHEL8 repos, get Ohsnap repofile, and enable and download satellite
+    sat_ready_rhel.register_to_cdn()
+    sat_ready_rhel.download_repofile(product='satellite', release=settings.server.version.release)
+    sat_ready_rhel.execute('dnf -y module enable satellite:el8 && dnf -y install satellite')
+    installed_version = sat_ready_rhel.execute('rpm -qa satellite').stdout
+    assert sat_version in installed_version
+    # Install Satellite
+    result = sat_ready_rhel.execute(
+        InstallerCommand('scenario satellite').get_command(), timeout='30m'
```

Comment:
I'm pretty sure the failure will be caught at the `result.status` line, but the log will have output from the installer run. I can implement what @ekohl is saying and instead check logs.

Reply:
What part of the log do you think I should be checking for @ekohl ?

Reply:
I don't have an installer log that failed at hand, but you can see something from a passed run:
```
2023-03-28 21:08:52 [DEBUG ] [configure] /Stage[main]/Puppet::Server::Config/Exec[migrate Puppetserver cadir]: Starting to evaluate the resource (309 of 1450)
2023-03-28 21:08:52 [DEBUG ] [configure] /Stage[main]/Puppet::Server::Config/Exec[migrate Puppetserver cadir]/creates: Checking that 'creates' path '/etc/puppetlabs/puppetserver/ca' exists
2023-03-28 21:08:52 [DEBUG ] [configure] Exec[migrate Puppetserver cadir](provider=posix): Executing check 'test -d '/etc/puppetlabs/puppet/ssl/ca' && ! test -L '/etc/puppetlabs/puppet/ssl''
2023-03-28 21:08:52 [DEBUG ] [configure] Executing: 'test -d '/etc/puppetlabs/puppet/ssl/ca' && ! test -L '/etc/puppetlabs/puppet/ssl''
2023-03-28 21:08:52 [DEBUG ] [configure] /Stage[main]/Puppet::Server::Config/Exec[migrate Puppetserver cadir]: '/opt/puppetlabs/bin/puppetserver ca migrate' won't be executed because of failed check 'onlyif'
2023-03-28 21:08:52 [DEBUG ] [configure] /Stage[main]/Puppet::Server::Config/Exec[migrate Puppetserver cadir]/creates: Checking that 'creates' path '/etc/puppetlabs/puppetserver/ca' exists
2023-03-28 21:08:52 [DEBUG ] [configure] Exec[migrate Puppetserver cadir](provider=posix): Executing check 'test -d '/etc/puppetlabs/puppet/ssl/ca' && ! test -L '/etc/puppetlabs/puppet/ssl''
2023-03-28 21:08:52 [DEBUG ] [configure] Executing: 'test -d '/etc/puppetlabs/puppet/ssl/ca' && ! test -L '/etc/puppetlabs/puppet/ssl''
2023-03-28 21:08:52 [DEBUG ] [configure] /Stage[main]/Puppet::Server::Config/Exec[migrate Puppetserver cadir]: '/opt/puppetlabs/bin/puppetserver ca migrate' won't be executed because of failed check 'onlyif'
2023-03-28 21:08:52 [INFO  ] [configure] /Stage[main]/Puppet::Server::Config/Exec[migrate Puppetserver cadir]: Triggered 'refresh' from 1 event
2023-03-28 21:08:52 [DEBUG ] [configure] /Stage[main]/Puppet::Server::Config/Exec[migrate Puppetserver cadir]: The container Class[Puppet::Server::Config] will propagate my refresh event
2023-03-28 21:08:52 [DEBUG ] [configure] /Stage[main]/Puppet::Server::Config/Exec[migrate Puppetserver cadir]: Evaluated in 0.02 seconds
```

If an exec resource failed, that's what you want to see. A common one to fail is `foreman-rake db:migrate` and the command output usually explains where it happened by giving you a stack trace.

Now it is hard to extract this from the log right now. A common way is to look for `[ERROR` and then see what's below that but it's hardly reliable so say `grep '[ERROR' --after-context=10` because 10 lines could be too little. Perhaps 50 or 100 lines would be a safe bet?

This is why I want to improve the installer to do this properly, but in the mean time we'll have to resort to getting the log file.